test/e2e/kubectl/kubectl.go

```diff
@@ -2069,6 +2069,103 @@ metadata:
 		})
 	})
+
+	ginkgo.Describe("Kubectl rollout", func() {
+		deploymentYaml := `
+		{
+			"apiVersion": "apps/v1",
+			"kind": "Deployment",
+			"metadata": {
+			  "name": "nginx-deployment"
+			},
+			"spec": {
+			  "replicas": 1,
+			  "selector": {
+				"matchLabels": {
+				  "test": "rollout"
+				}
+			  },
+			  "template": {
+				"metadata": {
+				  "labels": {
+					"test": "rollout"
+				  }
+				},
+				"spec": {
+				  "containers": [
+					{
+					  "name": "nginx-container",
+					  "image": "nginx:latest"
+					}
+				  ]
+				}
+			  }
+			}
+		}	  
+`
+		ginkgo.AfterEach(func() {
+			cleanupKubectlInputs(deploymentYaml, ns, "test=rollout")
+		})
+		ginkgo.It("update deployment env and rollback", func(ctx context.Context) {
+			// create deployment
+			if _, err := e2ekubectl.RunKubectlInput(ns, deploymentYaml, "create", "-f", "-"); err != nil {
+				framework.Failf("kubectl failed to create deployment")
+			}
+			if err := wait.PollUntilContextCancel(ctx, 10*time.Second, true, func(ctx context.Context) (done bool, err error) {
+				deployment, err := c.AppsV1().Deployments(ns).Get(ctx, "nginx-deployment", metav1.GetOptions{})
+				if err != nil {
+					return false, err
+				}
+				readyReplicas := deployment.Status.ReadyReplicas
+				desiredReplicas := deployment.Spec.Replicas
+				return readyReplicas == *desiredReplicas, nil
+			}); err != nil {
+				framework.Failf("created deployment not ready")
+			}
+
+			// do a small update
+			if _, err := e2ekubectl.RunKubectl(ns, "set", "env", "deployment/nginx-deployment", "foo=bar"); err != nil {
+				framework.Failf("kubectl failed set env for deployment")
+			}
+			// wait for env to be set
+			if err := wait.PollUntilContextCancel(ctx, 10*time.Second, true, func(ctx context.Context) (done bool, err error) {
+				deployment, err := c.AppsV1().Deployments(ns).Get(ctx, "nginx-deployment", metav1.GetOptions{})
+				if err != nil {
+					return false, err
+				}
+				readyReplicas := deployment.Status.ReadyReplicas
+				desiredReplicas := deployment.Spec.Replicas
+				if readyReplicas != *desiredReplicas {
+					return false, nil
+				}
+				envs := deployment.Spec.Template.Spec.Containers[0].Env
+				for _, env := range envs {
+					if env.Name == "foo" && env.Value == "bar" {
+						return true, nil
+					}
+				}
+				return false, nil
+			}); err != nil {
+				framework.Failf("can't find env in updated deployment")
+			}
+
+			// rollback
+			if _, err := e2ekubectl.RunKubectl(ns, "rollout", "undo", "deployment/nginx-deployment"); err != nil {
+				framework.Failf("kubectl failed to rollback deployment")
+			}
+			// wait for env to be reset
+			if err := wait.PollUntilContextCancel(ctx, 10*time.Second, true, func(ctx context.Context) (done bool, err error) {
+				deployment, err := c.AppsV1().Deployments(ns).Get(ctx, "nginx-deployment", metav1.GetOptions{})
+				if err != nil {
+					return false, err
+				}
+				envs := deployment.Spec.Template.Spec.Containers[0].Env
```

Comment:
can we also test that the deployment has only the original set of labels and annotations? To confirm this, it might be good to add some to the original yaml.

Reply:
I've updated the code to create a deployment yaml with label and annotations called `labeld-deployment.yaml.in` ,please take a look.